tasks/rubies.yml

```diff
@@ -1,17 +1,11 @@
 ---
 
-- name: Detect if rubies are installed
-  command: '{{ rvm1_rvm }} {{ item }} do true'
-  changed_when: False
-  failed_when: False
-  register: detect_rubies
-  with_items: '{{ rvm1_rubies }}'
-  when: rvm1_rubies is defined and rvm1_rubies | length > 0
-
 - name: Install rubies
-  command: '{{ rvm1_rvm }} install {{ item.item }} {{ rvm1_ruby_install_flags }}'
-  when: rvm1_rubies is defined and rvm1_rubies | length > 0 and item.rc|default(0) != 0
-  with_items: '{{ detect_rubies.results }}'
+  command: '{{ rvm1_rvm }} install {{ item }} {{ rvm1_ruby_install_flags }}'
```

Comment:
If I understand well, this will attempt to install all the rubies all the time, and rely on RVM idempotence, is that correct?

I believe this will work fine, but there is a big drawback which is time execution.

For instance in a testing environment here, with a real situation where there is currently 4 versions of Ruby, this will give (output snipped out for conciseness):

```
$ time for value in $(rvm list strings); do rvm install $value; done
Already installed ruby-2.5.3.
Already installed ruby-2.5.7.
Already installed ruby-2.5.8.
Already installed jruby-9.2.5.0.

real	0m2.388s
```

On the other hand, detecting installed rubies with another method gives:

```
$ time rvm list strings
ruby-2.5.3
ruby-2.5.7
ruby-2.5.8
jruby-9.2.5.0

real	0m0.230s
```

So in this case, on every run for machines where the configuration is already installed, this task will now take 10 times longer.

Also, re-installing something already installed could lead to unexpected behaviour at time, if I'm a bit more precautious (although this is probably won't lead to actual problems too often in that case).

Ideally, I think I would use `rvm list strings` then split to detect installed rubies initially, then install only on (`rvm1_rubies` - `detected_rubies`).

I believe this can be achieved with the `difference` filter:

http://docs.ansible.com/ansible/playbooks_filters.html#set-theory-filters

Reply:
@thbar your understanding seems correct. And I think that's a valid point about the overall time difference. Thanks for testing and catching that. I will update it to use `rvm list strings` and run `rmv install` on the missing rubies.